test/soap.test.ts

```diff
@@ -88,9 +88,10 @@ describe('convert and merge', () => {
       conn370.metadata.read('CustomField', 'Lead.Status'),
     ]);
     leadIds = rets.map((r) => r.id!);
-    convertedStatus = (statusField as any).picklist.picklistValues
-      .filter((pv: any) => pv.converted === 'true')
-      .map((pv: any) => pv.fullName)[0];
+    // @ts-expect-error  @typescript-eslint/no-unnecessary-type-assertion weird CI only linting?
+    convertedStatus = statusField.picklist.picklistValues
```

Comment:
statusField was being casted before:
`(statusField as any)`

probably the outdated MDAPI types.

Reply:
yeah - it was throwing an "unnecessary type assertion" error [before] (https://github.com/jsforce/jsforce/actions/runs/8976679543/job/24653986013) with `as any`, but then in the editor it was passing